### PR TITLE
Added .gitignore with firmware and python build file ignores

### PR DESCRIPTION
**Pull request by dodgymike**
Added .gitignore with firmware and python build file ignores

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+**/*.pyc
```
Comment (atlas0fd00m) [inline .gitignore:1]: this chunk is fine :) and in line with the PR title.

File: firmware/Makefile [hunk 2]
```diff
@@ -2,6 +2,7 @@
 BOOTLOADER_SIZE = 0x1400
 LDFLAGS_FLASH = --code-loc $(BOOTLOADER_SIZE)
 USB_DEVICE_SERIAL_NUMBER="`./new_serial.py`"
+DEVICE_SERIAL_NUMBER="`cat .serial`"
```
Comment (atlas0fd00m) [inline firmware/Makefile:5]: why the additional tracking of the serial#?  one serial number per device is preferred.  it should be the same one for usb as it is for the firmware check.

File: firmware/Makefile [hunk 3]
```diff
@@ -31,7 +32,7 @@ ys1fw: RfCatYS1
 # helper targets 
 chipcon_usb.rel: chipcon_usb.c include/*.h
 	#./new_serial.py
-	$(CC) $(CFLAGS) $(DEFINES)  -DUSB_DEVICE_SERIAL_NUMBER=$(USB_DEVICE_SERIAL_NUMBER) -c chipcon_usb.c
+	$(CC) $(CFLAGS) $(DEFINES)  -DUSB_DEVICE_SERIAL_NUMBER=$(USB_DEVICE_SERIAL_NUMBER) -DDEVICE_SERIAL_NUMBER=$(DEVICE_SERIAL_NUMBER) -c chipcon_usb.c
```
Comment (atlas0fd00m) [inline firmware/Makefile:35]: again, shouldn't be necessary.

File: firmware/Makefile [hunk 4]
```diff
@@ -343,9 +344,13 @@ installRfCatDonsDongle: RfCatDons.hex
 
 installRfCatDonsDongleCCBootloader: RfCatDonsCCBootloader.hex
 	if [ ! -c /dev/RFCAT_BL_D ] ; then ../rfcat --bootloader --force && sleep 1 ; fi ;
-	rfcat_bootloader /dev/RFCAT_BL_D erase_all
-	rfcat_bootloader /dev/RFCAT_BL_D download bins/RfCatDonsCCBootloader.hex
-	rfcat_bootloader /dev/RFCAT_BL_D verify bins/RfCatDonsCCBootloader.hex && rfcat_bootloader /dev/RFCAT_BL_D run
+#	rfcat_bootloader /dev/RFCAT_BL_D erase_all
+#	rfcat_bootloader /dev/RFCAT_BL_D download bins/RfCatDonsCCBootloader.hex
+#	rfcat_bootloader /dev/RFCAT_BL_D verify bins/RfCatDonsCCBootloader.hex && rfcat_bootloader /dev/RFCAT_BL_D run
+	./rfcat_bootloader /dev/ttyACM0 erase_all
```
Comment (atlas0fd00m) [inline firmware/Makefile:350]: please install the udev config file as instructed in README.  this change will is unnecessary

File: firmware/chipcon_usb.c [hunk 5]
```diff
@@ -46,6 +45,8 @@ __xdata int (*cb_ep0out)(void);
 __xdata int (*cb_ep0vendor)(USB_Setup_Header* __xdata );
 __xdata int (*cb_ep5)(void);
 
+__code u8 deviceserialnumber[] = QUOTE(DEVICE_SERIAL_NUMBER);
```
Comment (atlas0fd00m) [inline firmware/chipcon_usb.c:48]: unnecessary.  please use the USB serial#

File: firmware/chipcon_usb.c [hunk 6]
```diff
@@ -1168,6 +1169,10 @@ void processOUTEP5(void)
                 appReturn(2, ptr);
                 break;
 
+            case CMD_DEVICE_SERIAL_NUMBER:
```
Comment (atlas0fd00m) [inline firmware/chipcon_usb.c:1172]: make this point to the usb serial number (stored just after the bootloader and before the beginning of firmware) and we're gtg.

**Review by atlas0fd00m — CHANGES_REQUESTED** [review]

File: firmware/Makefile (hunk 4 above)
Comment (dodgymike) [reply firmware/Makefile:350]: Ooops, accidentally included this! Will remove it

**Comment by dodgymike** [comment]
Apologies - I was lazy and didn't make branches for each change. Will close this and do it properly!

**Pull request closed without merging**